data_mgr.py

```diff
@@ -0,0 +1,290 @@
+        else:
+            nodes = (
+                n for w_id in set(w_ids)
+                for n in getattr(self.data[w_id], node_type, []))
+        ids = [self.parse_node_id(n_id, node_type) for n_id in args['ids']]
+        exids = [self.parse_node_id(n_id, node_type) for n_id in args['exids']]
+        result = []
+        for node in nodes:
+            if ((not nat_ids or node.id in set(nat_ids)) and
+                    self._node_filter(node, ids, exids, args)):
+                result.append(node)
+        return result
+
+    def get_node_id(self, node_type, args):
+        """Return GrapphQL task object for given id."""
+        nat_id = args.get('native_ids', None)
+        oname, wname, nid = nat_id.split('/', 2)
+        w_id = f"{oname}/{wname}"
+        if node_type == 'proxy_nodes':
+            nodes = (
+                list(getattr(self.data[w_id], 'task_proxies', []))
+                + list(getattr(self.data[w_id], 'family_proxies', [])))
+        else:
+            nodes = getattr(self.data[w_id], node_type, [])
+        for node in nodes:
+            if node.id == nat_id:
+                return node
+        return None
+
+    def get_edges_all(self, args):
+        w_args = {
+            'ids': args['workflows'],
+            'exids': args['exworkflows'],
+        }
+        result = []
+        for edge in [
+                n for k in self.get_workflow_msgs(w_args)
+                for n in getattr(k, 'edges')]:
+            result.append(edge)
+        return result
+
+    def get_edges_id(self, args):
+        nat_ids = args.get('native_ids', [])
+        w_ids = []
+        for nat_id in nat_ids:
+            oname, wname, eid = nat_id.split('/', 2)
+            w_ids.append(f"{oname}/{wname}")
+        w_ids = list(set(w_ids))
+        result = []
+        for edge in [
+                e for k in w_ids
+                for e in getattr(self.data[k], 'edges')]:
+            result.append(edge)
+        return result
+
+    def _node_filter(self, node, ids, exids, args):
+        natts = [
+            getattr(node, 'cycle_point', None),
+            getattr(node, 'namespace', [
+                n for n in [getattr(node, 'name', None)] if n is not None]),
+            getattr(node, 'submit_num', None),
+            getattr(node, 'state', None),
+        ]
+        return (
+                (not args.get('states') or node.state in args['states']) and
+                not (
+                    args.get('exstates') and
+                    node.state in args['exstates']) and
+                (args.get('mindepth', -1) < 0 or
+                    node.depth >= args['mindepth']) and
+                (args.get('maxdepth', -1) < 0 or
+                    node.depth <= args['maxdepth']) and
+                (not ids or self._node_atts_filter(natts, ids)) and
+                not (exids and self._node_atts_filter(natts, exids)))
+
+    @staticmethod
```

Comment:
I guess that would make sense if they weren't only used/had-utility as helper functions to other methods in the class and were not related to the theme of the class. If any of them qualify it would be `parse_workflow_id` and `parse_node_id`..

Reply:
Only the two mentioned.